test/DelegateCompensationStaker.t.sol

```diff
@@ -80,3 +80,335 @@ contract InitializeDelegateCompensation is DelegateCompensationStakerTest {
     staker.initializeDelegateCompensation(_delegate);
   }
 }
+
+contract SetAdmin is DelegateCompensationStakerTest {
+  function testFuzz_AdminCanSetNewAdmin(address _newAdmin) public {
+    vm.assume(_newAdmin != address(0));
+
+    vm.expectEmit();
+    emit Staker.AdminSet(admin, _newAdmin);
+
+    vm.prank(admin);
+    staker.setAdmin(_newAdmin);
+
+    assertEq(staker.admin(), _newAdmin);
+  }
+}
+
+contract SetEarningPowerCalculator is DelegateCompensationStakerTest {
+  function testFuzz_AdminCanSetEarningPowerCalculator(address _newCalculator) public {
+    vm.assume(_newCalculator != address(0));
+
+    vm.expectEmit();
+    emit Staker.EarningPowerCalculatorSet(address(calculator), _newCalculator);
+
+    vm.prank(admin);
+    staker.setEarningPowerCalculator(_newCalculator);
+
+    assertEq(address(staker.earningPowerCalculator()), _newCalculator);
+  }
+}
+
+contract SetMaxBumpTip is DelegateCompensationStakerTest {
+  function testFuzz_AdminCanSetMaxBumpTip(uint256 _newMaxBumpTip) public {
+    vm.expectEmit();
+    emit Staker.MaxBumpTipSet(MAX_BUMP_TIP, _newMaxBumpTip);
+
+    vm.prank(admin);
+    staker.setMaxBumpTip(_newMaxBumpTip);
+
+    assertEq(staker.maxBumpTip(), _newMaxBumpTip);
+  }
+}
+
+contract SetRewardNotifier is DelegateCompensationStakerTest {
+  function testFuzz_AdminCanSetRewardNotifier(address _notifier, bool _isEnabled) public {
+    vm.expectEmit();
+    emit Staker.RewardNotifierSet(_notifier, _isEnabled);
+
+    vm.prank(admin);
+    staker.setRewardNotifier(_notifier, _isEnabled);
+
+    assertEq(staker.isRewardNotifier(_notifier), _isEnabled);
+  }
+}
+
+contract SetClaimFeeParameters is DelegateCompensationStakerTest {
+  function testFuzz_AdminCanSetClaimFeeParameters(
+    DelegateCompensationStaker.ClaimFeeParameters memory _newParams
+  ) public {
+    vm.assume(_newParams.feeCollector != address(0));
+    _newParams.feeAmount = uint96(bound(_newParams.feeAmount, 0, staker.MAX_CLAIM_FEE()));
+
+    (uint96 _initialFeeAmount, address _initialFeeCollector) = staker.claimFeeParameters();
+
+    vm.expectEmit();
+    emit Staker.ClaimFeeParametersSet(
+      _initialFeeAmount, _newParams.feeAmount, _initialFeeCollector, _newParams.feeCollector
+    );
+
+    vm.prank(admin);
+    staker.setClaimFeeParameters(_newParams);
+
+    (uint96 _newFeeAmount, address _newFeeCollector) = staker.claimFeeParameters();
+    assertEq(_newFeeAmount, _newParams.feeAmount);
+    assertEq(_newFeeCollector, _newParams.feeCollector);
+  }
+}
+
+contract LastTimeRewardDistributed is DelegateCompensationStakerTest {
```

Comment:
We can probably remove this and RewardPerTokenAccumulated. These are used within other functions so they should be under test.